core/src/test/java/io/aiven/kafka/tieredstorage/fetch/index/MemorySegmentIndexesCacheTest.java

```diff
@@ -0,0 +1,291 @@
+                IndexType.TIMESTAMP,
+                timeIndexSupplier
+            );
+            assertThat(cachedTimeIndex).hasBinaryContent(TIME_INDEX);
+            verifyNoMoreInteractions(timeIndexSupplier);
+            assertThat(cache.cache.asMap()).hasSize(2);
+
+            verifyNoMoreInteractions(removalListener);
+        }
+
+        @Test
+        void timeBasedEviction() throws IOException, StorageBackendException, InterruptedException {
+            cache.configure(Map.of(
+                "size", "-1",
+                "retention.ms", "100"
+            ));
+            assertThat(cache.cache.asMap()).hasSize(0);
+
+            final ObjectKey key = objectKeyFactory.key(REMOTE_LOG_SEGMENT_METADATA, ObjectKeyFactory.Suffix.INDEXES);
+            final InputStream offsetIndex = cache.get(
+                key,
+                IndexType.OFFSET,
+                offsetIndexSupplier
+            );
+            assertThat(offsetIndex).hasBinaryContent(OFFSET_INDEX);
+            assertThat(cache.cache.asMap()).hasSize(1);
+            final InputStream cachedOffsetIndex = cache.get(
+                key,
+                IndexType.OFFSET,
+                offsetIndexSupplier
+            );
+            assertThat(cachedOffsetIndex).hasBinaryContent(OFFSET_INDEX);
+            verifyNoMoreInteractions(offsetIndexSupplier);
+            assertThat(cache.cache.asMap()).hasSize(1);
+
+            // Wait enough for the cache to expire the entry. Based on retention.ms.
+            Thread.sleep(100);
+
+            final InputStream timeIndex = cache.get(
+                key,
+                IndexType.TIMESTAMP,
+                timeIndexSupplier
+            );
+            assertThat(timeIndex).hasBinaryContent(TIME_INDEX);
+            assertThat(cache.cache.asMap().size()).isGreaterThanOrEqualTo(1);
+            final InputStream cachedTimeIndex = cache.get(
+                key,
+                IndexType.TIMESTAMP,
+                timeIndexSupplier
+            );
+            assertThat(cachedTimeIndex).hasBinaryContent(TIME_INDEX);
+            verifyNoMoreInteractions(timeIndexSupplier);
+            assertThat(cache.cache.asMap().size()).isGreaterThanOrEqualTo(1);
+
+            await().atMost(Duration.ofMillis(5000)).pollInterval(Duration.ofMillis(100))
+                .until(() -> !mockingDetails(removalListener).getInvocations().isEmpty());
+
+            assertThat(cache.cache.asMap()).hasSize(1);
+            verify(removalListener)
+                .onRemoval(
+                    argThat(argument -> argument.indexType == IndexType.OFFSET),
+                    any(),
+                    eq(RemovalCause.EXPIRED));
+        }
+
+        @Test
+        void sizeBasedEviction() throws IOException, StorageBackendException {
+            cache.configure(Map.of(
+                "size", "18",
+                "retention.ms", "-1"
+            ));
+            assertThat(cache.cache.asMap()).hasSize(0);
```

Comment:
`isEmpty()`? and in other places. Also perhaps `isNotEmpty()` for `.size()).isGreaterThanOrEqualTo(1)`?

Reply:
I see you fixed it here but what about other places?

Reply:
Found it. Should be fixed on the latest commit